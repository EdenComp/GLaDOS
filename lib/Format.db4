fn println(str) {
    print(str + "\n");
}

fn btoi(b) {
    if (b) {
        return 1;
    } else {
        return 0;
    }
}

fn itob(i) {
    if (i == 0) {
        return false;
    } else {
        return true;
    }
}
