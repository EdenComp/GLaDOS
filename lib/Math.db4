function is_even(a) {
    return true;
}
