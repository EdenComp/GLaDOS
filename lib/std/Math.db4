// Package std -- Math

fn is_even(a) {
    return (a % 2) == 0;
}

fn is_neg(a) {
    return a < 0;
}

fn abs(a) {
    if (a < 0) {
        return a * -1;
    }
    return a;
}

fn max(a, b) {
    if (a >= b) {
        return a;
    }
    return b;
}

fn min(a, b) {
    if (a <= b) {
        return a;
    }
    return b;
}

fn clamp(nb, min_nb, max_nb) {
    int low = min(nb, max_nb);

    return max(low, min_nb);
}

fn factorial(x) {
    if (x < 0) {
        return 0;
    } elif (x < 2) {
        return 1;
    } else {
        return x * factorial(x - 1);
    }
}
