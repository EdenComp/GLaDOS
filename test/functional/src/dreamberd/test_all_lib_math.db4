import "std:Format";
import "std:Math";

int randomvarnamesoitssafehaha = is_even(2);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = is_even(3);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = is_neg(-2);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = is_neg(3);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = abs(-2);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = abs(3);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = min(2, 3);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = min(3, 2);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = max(2, 3);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = max(3, 2);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = clamp(2, 3, 4);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = clamp(3, 2, 4);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = clamp(5, 2, 4);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = clamp(1, 2, 4);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = clamp(3, 2, 4);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = clamp(4, 2, 4);
print(randomvarnamesoitssafehaha);

randomvarnamesoitssafehaha = factorial(5);
print(randomvarnamesoitssafehaha);
randomvarnamesoitssafehaha = factorial(0);
print(randomvarnamesoitssafehaha);
