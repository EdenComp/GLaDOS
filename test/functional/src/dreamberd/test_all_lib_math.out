TrueFalseTrueFalse2322333342341201